torch/csrc/jit/serialization/python_print.cpp

```diff
@@ -1180,6 +1182,28 @@ struct PythonPrintImpl {
           }
           stmt << *v;
         }
+
+        // scan backwards to find the first index where out args start
+        size_t out_start = num_schema_args - 1;
+        while (out_start >= necessary_args) {
+          auto arg = schema.arguments().at(out_start);
+          if (!arg.is_out()) {
+            break;
+          }
+          out_start--;
+        }
+
+        // print out args
+        for (size_t i = out_start + 1; i < schema.arguments().size(); i++) {
+          stmt << ", ";
+          auto arg = schema.arguments().at(i);
+          TORCH_INTERNAL_ASSERT(arg.is_out());
+          // figure out the corresponding input at this index
+          auto input_idx = node->inputs().size() - (num_schema_args - i);
+          if (input_idx < node->inputs().size()) {
```

Comment:
why is this IF check needed?

Reply:
I think there could be cases where actual provided inputs are less/more than the number of arguments in the schema due to star sign in the schema. In that case, I think there will index mismatch so only reliable way to tell out args from the input would be assuming end inputs are out.